### PR TITLE
Add e2e tests to check new_pusher compatibility

### PR DESCRIPTION
**Pull request by xiaohegong**
Added e2e tests for checking compatibility of new pusher.

### CODE REVIEW

File: tests/container/BUILD [hunk 1]
```diff
@@ -320,6 +320,21 @@ container_push(
...
+    name = "alpine_with_custom_attr",
+    base = "@new_alpine_linux_amd64//image",
+    cmd = ["Hello World!"],
+    entrypoint = ["echo"],
+    files = [
+        "//testdata:bar",
+        "//testdata:foo",
+    ],
+    layers = [
+        "//testdata:env_layer",
```
Comment (smukherj1) [inline tests/container/BUILD]: Can you add another layer here that has files? I want to ensure the pusher works when there's more than one layer with files to upload. Also, for this case we want to run a `container_pull` followed by `container_test` to ensure files from both layers were uploaded to the registry correctly.

File: tests/container/BUILD [hunk 2]
```diff
@@ -370,20 +385,70 @@ new_container_push(
...
     format = "legacy",
+    image = ":alpine_with_custom_attr",
+    registry = "localhost:5000",
+    repository = "docker/test",
+    tag = "test",
+    tags = ["manual"],
+)
+
+new_container_push(
+    name = "new_push_test_legacy_auth",
```
Comment (xingao267) [inline tests/container/BUILD]: Is the image also from container_img? If so, make the naming consistent.

File: tests/container/BUILD [hunk 3]
```diff
@@ -370,20 +385,70 @@ new_container_push(
...
+    format = "legacy",
     image = ":set_cmd_and_entrypoint",
     registry = "localhost:5000",
     repository = "docker/test",
     tag = "test",
     tags = ["manual"],
 )
 
+new_container_push(
+    name = "new_push_test_pulled_legacy",
```
Comment (xingao267) [inline tests/container/BUILD]: new_push_test_legacy_from_old_puller

File: tests/container/BUILD [hunk 4]
```diff
@@ -370,20 +385,70 @@ new_container_push(
     tags = ["manual"],
 )
 
+new_container_push(
+    name = "new_push_test_pulled_oci",
```
Comment (xingao267) [inline tests/container/BUILD]: new_push_test_oci_from_new_puller

File: tests/container/BUILD (hunk 2 above)
Comment (xingao267) [reply tests/container/BUILD]: like new_push_test_legacy_from_container_img_with_auth

File: tests/container/BUILD [hunk 5]
```diff
@@ -370,20 +385,70 @@ new_container_push(
...
+    format = "legacy",
+    image = "@new_alpine_linux_ppc64le//image",
+    registry = "localhost:5000",
+    repository = "docker/test",
+    tag = "test",
+    tags = ["manual"],
+)
+
+new_container_push(
+    name = "new_push_test_old_puller_compat",
```
Comment (xingao267) [inline tests/container/BUILD]: new_push_test_legacy_from_old_puller

**Pull request merged**